### PR TITLE
Partially working event generation

### PR DESCRIPTION
**Pull request by cquiroz**
This is a first attempt to get the events integrated into observe

The structure of the events on the ODB is a bit counter to the observe engine design and there are still problems to store the step events as we need to store certain ids along the way

Some of the parts failing are commented in this PR

At this moment we can store visit, records and atoms but not steps yet

### CODE REVIEW

File: modules/server_new/src/main/scala/observe/server/ObserveActions.scala [hunk 1]
```diff
@@ -121,9 +116,7 @@ trait ObserveActions {
     env:    ObserveEnvironment[F]
   ): F[ObserveCommandResult] =
     for {
-      _ <- env.ctx.visitId
-             .map(sendDataStart(env.odb, _, env.obsId, env.stepId, env.datasetIndex, fileId))
-             .getOrElse(Applicative[F].unit)
+      // _ <- sendDataStart(env.odb, env.obsId, RecordedStepId(null), fileId)
```
Comment (rpiaggio) [inline modules/server_new/src/main/scala/observe/server/ObserveActions.scala:119]: commented code

File: modules/server_new/src/main/scala/observe/server/ObserveEngine.scala [hunk 2]
```diff
@@ -421,17 +425,23 @@ object ObserveEngine {
...
+                                         seq.seqGen.sequenceType,
+                                         NonNegShort.unsafeFrom(seq.seqGen.steps.length.toShort),
+                                         seq.seqGen.staticCfg
+                          )
                           .map { i =>
                             Event.modifyState {
                               executeEngine
-                                .modify(
+                                .modify {
+                                  println("MOTIDY")
```
Comment (rpiaggio) [inline modules/server_new/src/main/scala/observe/server/ObserveEngine.scala:438]: debug code

**Review by rpiaggio — APPROVED** [review]
LGTM

**Review by jluhrs — APPROVED** [review]
LGTM

**Pull request merged**